src/store/index.ts

```diff
@@ -0,0 +1,91 @@
+import Vue from 'vue';
+import Vuex, { Store } from 'vuex';
+import { createDirectStore } from 'direct-vuex';
+
+import api from '@/api';
+
+Vue.use(Vuex);
+
+interface WorkspaceSpec {
+  name: string;
+  nodeTables?: string[];
+  edgeTables?: string[];
+  graphs?: string[];
+}
+
+interface State {
+  workspaces: string[];
+  currentWorkspace: WorkspaceSpec | null;
+}
+
+const {
+  store,
+  rootActionContext,
+  moduleActionContext,
+  rootGetterContext,
+  moduleGetterContext,
+} = createDirectStore({
+  state: {
+    workspaces: [],
+    currentWorkspace: null,
+  } as State,
+  getters: {
+    nodeTables(state: State): string[] {
+      if (state.currentWorkspace !== null && state.currentWorkspace.nodeTables) {
+        return state.currentWorkspace.nodeTables;
+      }
+      return [];
+    },
+    edgeTables(state: State): string[] {
+      if (state.currentWorkspace !== null && state.currentWorkspace.edgeTables) {
+        return state.currentWorkspace.edgeTables;
+      }
+      return [];
+    },
+    graphs(state: State) {
+      if (state.currentWorkspace !== null && state.currentWorkspace.graphs) {
+        return state.currentWorkspace.graphs;
+      }
+      return [];
+    },
+  },
+  mutations: {
+    setWorkspaces(state, workspaces: string[]) {
+      state.workspaces = workspaces;
+    },
+    setCurrentWorkspace(state, workspace: WorkspaceSpec) {
+      state.currentWorkspace = workspace;
+    },
+  },
+  actions: {
+    async fetchWorkspaces(context) {
+      const { commit } = rootActionContext(context);
+      const workspaces = await api.workspaces();
+      commit.setWorkspaces(workspaces);
+    },
+    async fetchWorkspace(context, workspace: string) {
+      const { commit } = rootActionContext(context);
+      const nodeTables = await api.tables(workspace, { type: 'node' });
+      const edgeTables = await api.tables(workspace, { type: 'edge' });
+      const graphs = await api.graphs(workspace);
+
+      commit.setCurrentWorkspace({ name: workspace, nodeTables, edgeTables, graphs });
+    },
```

Comment:
It might be worthwhile doing `commit.setCurrentWorkspace({ name: workspace, [], [], [] })`  first, because when you switch workspaces at the moment, there is a little lag where the old names show. Setting everything to empty will at least not show the old networks and tables as you're moving around

Reply:
Good catch, just tried it locally and it makes the UI feel much smoother. I'll push soon.